MenuButton.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  MenuButton.swift
+//  Sentinel
+//
+//  Created by Infinum on 12.09.2023..
+//
+
+import UIKit
+
+/// A button that opens a context menu with enum cases (raw representables) as options.
+class MenuButton: UIButton {
+
+    private var _enumCases: [any RawRepresentable] = []
```

Comment:
Let's ditch the underscore naming for private properties to keep things in line with our code style 🙂